### PR TITLE
Use `displayNames` in UI App Selector Cards

### PR DESCRIPTION
**Pull request by evantahler**
This PR continues https://github.com/grouparoo/grouparoo/pull/2342, and uses the displayNames in the App Selector cards in the UI

Screen Shot 2021-09-30 at 3 08 09 PM

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

N/A

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/src/modules/plugins.ts [hunk 1]
```diff
@@ -42,6 +42,17 @@ export namespace Plugins {
...
+    name: string;
+    description: string;
+    imageUrl: string;
+    packageName: string;
+    source: boolean;
+    destination: boolean;
+    npmUrl?: string;
+    docsUrl?: string;
+  };
+
```
Comment (evantahler) [inline core/src/modules/plugins.ts:55]: Finally typing the plugin data we get from https://www.grouparoo.com/plugins/v1/manifest.json

File: ui/ui-components/components/appSelectorList.tsx [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
 import { CardDeck } from "react-bootstrap";
 import { useRouter } from "next/router";
+import { Actions } from "../utils/apiData";
+
+type SelectableApp = Actions.AppOptions["types"][number];
+type SelectablePackage = Actions.PluginsAvailableList["plugins"][number];
+type SelectableSource =
+  Actions.SourceConnectionApps["connectionApps"][number]["app"];
+type SelectableDestination =
+  Actions.DestinationConnectionApps["connectionApps"][number]["app"];
```
Comment (evantahler) [inline ui/ui-components/components/appSelectorList.tsx:11]: This component was a bit wild before - handling 3 types of API responses without any typing.... now we have types!

File: ui/ui-components/components/appSelectorList.tsx [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+
+function isSelectableApp(
+  item:
+    | SelectableApp
+    | SelectablePackage
+    | SelectableSource
+    | SelectableDestination
+): item is SelectableApp {
+  return (item as SelectableApp).plugin !== undefined;
+}
```
Comment (krishnaglick) [inline ui/ui-components/components/appSelectorList.tsx]: It's probably safe to do:

```suggestion
function isSelectableApp(
  item: any
): item is SelectableApp {
  return item.plugin !== undefined;
}
```

File: ui/ui-components/components/appSelectorList.tsx [hunk 4]
```diff
@@ -9,75 +37,107 @@ export default function AppSelectorList({
   displayAddAppButton = false,
 }: {
   onClick: Function;
-  items: Array<any>;
+  items:
+    | SelectableApp[]
+    | SelectablePackage[]
+    | SelectableSource[]
+    | SelectableDestination[];
```
Comment (krishnaglick) [inline ui/ui-components/components/appSelectorList.tsx:32]:
```suggestion
  items:
    (SelectableApp
    | SelectablePackage
    | SelectableSource
    | SelectableDestination)[];
```
Maybe?

Comment (evantahler) [reply ui/ui-components/components/appSelectorList.tsx:32]: It's an array of one type of thing - it won't be mixed

**Review by krishnaglick — APPROVED** [review]

**Pull request merged**